src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/InMemoryEventStore.java

```diff
@@ -99,11 +98,13 @@ public BatchPayload getEventsBatch(int numberToGet) {
      *
      * @param needRetry if true, move events back to the buffer instead of deleting
      * @param batchId the ID of the batch of events
+     * @return newer TrackerPayloads deleted from the queue to make space for older payloads
```

Comment:
Btw, just realizing now that I am a bit confused by this behaviour – Why would we remove newer events to make space for older events that failed to be sent to the collector? I would expect that the older events are dropped first and newer kept. No need to change this here, I just found it a bit weird.

Reply:
When the storage is full and new events arrive, they're dropped in favour of keeping the old ones that are already buffered. This is for consistency with that.
Why we chose to do that, I'm not sure. IIRC all the trackers have this behaviour.